### PR TITLE
Prompt default office selection after user signs in for the first time

### PR DESCRIPTION
**Pull request by ThuTrinh**
# Description
When a user signs in for the first time, their default office is set for them. This can cause confusion later down the track if they don't notice it.

Now, a welcome notification will appear with instructions to change the default office if they wish. The user can dismiss this notification and it won't appear again.

## References
[Github Issue](https://github.com/zendesk/volunteer_portal/issues/67)

## Implementation notes (if needed)
This PR also introduces:

### [Apollo Hooks](https://www.apollographql.com/docs/react/api/react-hooks/) 
Apollo Hooks Provider can live alongside the old ApolloProvider. It is a much more terse interface. Hopefully this can also help us with moving away from redux.

```javascript
import { useMutation } from '@apollo/react-hooks'
import ConfirmProfileSettingsMutation from './mutations/confirmProfileSettings.gql'

const ConfirmProfileAlert = (_props) => {
  const [confirmProfileSettings, _] = useMutation(ConfirmProfileSettingsMutation)

  return(<button onClick={() => confirmProfileSettings()}>Click Me</button>)
}
```

### UserContext
A custom context that stores the current logged in user, provided at login. Subsequent components can access this context through the `useContext` hook. No longer needing to relying on graphql queries or prop passing. This lets us decouple out components as well as minimise unnecessary queries.

```javascript
import React, { useContext } from 'react'
import { UserContext } from '../../context'

const UserInfo = () => {
  const currentUser = useContext(UserContext)

  return (<div>{currentUser.name}</div>)
}
```

## Screenshots
profile-setting-prompt

## CCs
@zendesk/volunteer

## Risks (if any)
* [low] - all additive

### CODE REVIEW

File: app/javascript/components/Notifications/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { Grid, Row, Col } from '@zendeskgarden/react-grid'
+import { ThemeProvider } from '@zendeskgarden/react-theming'
+import { UserContext } from '../../context'
+
+import ConfirmProfileAlert from './ConfirmProfileAlert'
+
+import '@zendeskgarden/react-grid/dist/styles.css'
+import '@zendeskgarden/react-notifications/dist/styles.css'
+
+const isFirstSignIn = R.propSatisfies(R.isNil, 'lastSignInAt')
```
Comment (samzx) [inline app/javascript/components/Notifications/index.js]: is there something we can do so user's that've already sign in before and have already changed their office won't get this prompt? It _may_ be noisy for those who've already set their office.

**Review by samzx — COMMENTED** [review]
The Apollo hooks and context are really nice 😍

File: app/javascript/components/Notifications/ConfirmProfileAlert.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  const [confirmProfileSettings, _] = useMutation(ConfirmProfileSettingsMutation)
+
+  const dismiss = () => {
+    confirmProfileSettings()
+    setShow(false)
+  }
+
+  if (show) {
+    return (
+      <Notification type="info">
```
[the remark this answers is not in the document]
Comment (ThuTrinh) [reply app/javascript/components/Notifications/ConfirmProfileAlert.js]: I was tossing between modal and notification. I don't really mind either way. The benefit of the modal is that it is more in your face :)

File: app/graphql/types/user_preference_graph_type.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module Types
+  class UserPreferenceGraphType < Types::BaseObject
+    graphql_name 'UserPreference'
+    description 'An office'
```
[the remark this answers is not in the document]
Comment (ThuTrinh) [reply app/graphql/types/user_preference_graph_type.rb]: woops

File: app/javascript/components/Notifications/index.js (hunk 1 above)
Comment (ThuTrinh) [reply app/javascript/components/Notifications/index.js]: hmm, I can backfill the user preference value for all existing users

**Review by samzx — APPROVED** [review]

**Pull request merged**